packages/backend/src/server.ts

```diff
@@ -14,18 +16,22 @@ import { generateSwaggerDocs } from './api-doc'
 import { EntityService } from './entity/service'
 import { SchemaService } from './schema/service'
 import { SubmissionService } from './submission/service'
+import { AuthController } from './auth/controller'
+import { authenticate } from './auth/middleware'
 
 const logger = createLogger(__filename)
 
 interface ServerConfig {
   port: number
   url: string
+  corsOriginUrl: string
 }
 
 interface Context {
   entityService: EntityService
   schemaService: SchemaService
   submissionService: SubmissionService
+  authController: AuthController
 }
```

Comment:
I'm not sure that I like the grouping of controllers and services together, perhaps we should have a service interface, and a controller interface?

Reply:
Maybe `Context` should look something like this?

```suggestion
interface Context {
  services: {
      entity: EntityService,
      schema: SchemaService,
      submission: SubmissionService
  },
  controllers: {
      auth: AuthController
  }
}
```

In order to group services / controllers together (as we intend to have more controllers)?

Reply:
Ah, I see. Yes, eventually we could updated it in that way. Ideally, I would like to build context more around modules. Similar to modules in AFJ for example. Module would have some public API (interface) and we would register just modules to the context. The module would enclose its own the controllers, services and repositories.

The structure you're suggesting is the opposite direction, where you organize code by layer rather than by functionality.

Reply:
Yeah, I'm happy with that also - i.e.

```suggestion
interface Context {
  entity: {
      service: EntityService
  },
  schema: {
      service: SchemaService
  },
  submission: {
      service: SubmissionService
  },
  auth: {
      controller: AuthController
  }
}
```